projects/ebv-analyzer-app/src/app/country-provider.service.ts

```diff
@@ -36,93 +35,53 @@ export class CountryProviderService {
-        // });
-        // const countrySourceOperator = new Operator({
-        //     operatorType: countryOperatorType,
-        //     resultType: ResultTypes.POLYGONS,
-        //     projection: Projections.WGS_84,
-        //     attributes: ['value'],
-        //     dataTypes: new Map<string, DataType>().set('tif_channel_id', DataTypes.Float32).set('NAME', DataTypes.Alphanumeric),
-        //     units: new Map<string, Unit>().set('tif_chn_id', Unit.defaultUnit).set('NAME', Unit.defaultUnit),
-        // });
-        // const newLayer = new VectorLayer({
-        //     name: country.name,
-        //     operator: countrySourceOperator,
-        //     symbology: VectorSymbology.createSymbology({
-        //         fillRGBA: {r: 0, g: 0, b: 0, a: 0},
-        //         strokeRGBA: {r: 255, g: 0, b: 0, a: 1},
-        //         strokeWidth: 3,
-        //     }),
-        // });
-        // if (this.vectorLayer) {
-        //     try {
-        //         this.projectService.removeLayer(this.vectorLayer);
-        //     } catch (e) {
-        //         // TODO: rule out that this can fail
-        //     }
-        // }
-        // this.projectService.addLayer(newLayer);
-        // this.vectorLayer = newLayer;
-    }
-
     public setSelectedCountry(country: Country): void {
         this.selectedCountry$.next(country);
-        this.replaceVectorLayerOnMap();
+
+        const workflow: WorkflowDict = {
+            type: 'Vector',
+            operator: {
+                type: 'OgrSource',
+                params: {
+                    dataset: COUNTRY_DATA_LIST[country.name].polygon,
+                },
+            },
+        };
+
+        this.projectService
+            .registerWorkflow(workflow)
+            .pipe(
+                mergeMap((workflowId) =>
+                    this.dataSelectionService.setPolygonLayer(
+                        new VectorLayer({
+                            workflowId,
+                            name: country.name,
+                            symbology: PolygonSymbology.fromPolygonSymbologyDict({
+                                type: 'polygon',
+                                stroke: {
+                                    width: {
+                                        type: 'static',
+                                        value: 2,
+                                    },
+                                    color: {
+                                        type: 'static',
+                                        color: [255, 0, 0, 255],
+                                    },
+                                },
+                                fillColor: {
+                                    type: 'static',
+                                    color: [0, 0, 0, 0],
+                                },
+                            }),
+                            isLegendVisible: false,
+                            isVisible: true,
+                        }),
```

Comment:
Two things:
* I guess we could use GEO BON green and maybe some very transparent filling?
* Maybe we can define it independently from this function

Reply:
in the current portal the outline is also red.